### PR TITLE
105 - ods lib component buttons

### PR DESCRIPTION
**Pull request by dolinetouko**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Types of change

- Bug fix (non-breaking which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Refactoring (non-breaking change)
- Breaking change (fix or feature that would change existing functionality)

#### Accessibility

- [x] My change follows accessibility good practices

#### Design

- [x] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- [x] My change is compatible with responsive display

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] Manually test on simulators and on a physical device (Android & iOS, orientation change, dark/light mode)
- [x] Code review
- [x] Design review
- [x] A11y review

### CODE REVIEW

File: app/lib/ui/components/buttons/component_elevated_button.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  @override
+  State<ComponentElevatedButtons> createState() => _ElevatedButtonsState();
+}
+
+class _ElevatedButtonsState extends State<ComponentElevatedButtons> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text('Elevated Buttons'),
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_elevated_button.dart:15]: rajouter les clefs l10n pour le texte 'Elevated Buttons'

File: app/lib/ui/components/buttons/component_elevated_button.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      child: IntrinsicWidth(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            ElevatedButton(
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_elevated_button.dart:46]: ce n'est pas le composant ODS qu'il faut utiliser ? **OdsElevatedButton**

File: app/lib/ui/components/buttons/component_filled_button.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  @override
+  State<ComponentFilledButtons> createState() => _ComponentFilledButtonsState();
+}
+
+class _ComponentFilledButtonsState extends State<ComponentFilledButtons> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text('Filled Buttons'),
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_filled_button.dart:16]: rajouter les clefs l10n pour le texte 'Filled Buttons'

File: app/lib/ui/components/buttons/component_filled_tonal_button.dart [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      _ComponentFilledTonalButtonsState();
+}
+
+class _ComponentFilledTonalButtonsState
+    extends State<ComponentFilledTonalButtons> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text('Filled Tonal Buttons'),
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_filled_tonal_button.dart:17]: rajouter les clefs l10n pour le texte 'Filled Tonal Buttons'

File: app/lib/ui/components/buttons/component_outlined_button.dart [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  State<ComponentOutlinedButtons> createState() =>
+      _ComponentOutlinedButtonsState();
+}
+
+class _ComponentOutlinedButtonsState extends State<ComponentOutlinedButtons> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: Text('Outlined Buttons'),
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_outlined_button.dart:17]: Filled Tonal Buttons

File: app/lib/ui/components/buttons/component_filled_tonal_button.dart [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      child: IntrinsicWidth(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            FilledButton.tonal(
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_filled_tonal_button.dart:48]: ce n'est pas le composant ODS qu'il faut utiliser ? **OdsFilledTonalButton**

File: app/lib/ui/components/buttons/component_outlined_button.dart [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      child: IntrinsicWidth(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            componentSpacer,
+            OutlinedButton(
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_outlined_button.dart:48]: ce n'est pas le composant ODS qu'il faut utiliser ? **OdsOutlinedButton**

File: app/lib/ui/components/buttons/component_text_button.dart [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      child: IntrinsicWidth(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            componentSpacer,
+            TextButton(
```
Comment (Tayebsed93) [inline app/lib/ui/components/buttons/component_text_button.dart:47]: ce n'est pas le composant ODS qu'il faut utiliser ? **OdsTextButton**

**Comment by Tayebsed93** [comment]
Désormais il est possible d'ajouter la customization en récupérant la branche main, c'est une étape supplémentaire à rajouter pour le composant bouton

**Comment by Tayebsed93** [comment]
> Désormais il est possible d'ajouter la customization en récupérant la branche main, c'est une étape supplémentaire à rajouter pour le composant bouton

OK en attente du design pour rajouter la personnalisation

File: app/CHANGELOG.md [hunk 9]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add `OdsVerticalImageFirstCard` widget demonstration ([#8](https://github.com/Orange-OpenSource/ods-flutter/issues/8))
 - Add recipes Json file and use it ([#123](https://github.com/Orange-OpenSource/ods-flutter/issues/123))
 - Add customization bottom sheet to configure components ([#148](https://github.com/Orange-OpenSource/ods-flutter/issues/148)))
```
Comment (ludovic35) [inline app/CHANGELOG.md:18]: remove last )

File: app/lib/ui/components/buttons/component_button_customization.dart [hunk 10]
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_button_customization.dart:1]: rename the file removing _component_ to_ have  _button_customization.dart_

File: app/lib/ui/components/buttons/component_elevated_button.dart [hunk 11]
```diff
@@ -0,0 +1,105 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_elevated_button.dart:1]: rename the file removing component to_ have elevated_button.dart

File: app/lib/ui/components/buttons/component_elevated_button.dart [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_gen/gen_l10n/ods_flutter_app_localizations.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:ods_flutter/components/button/ods_elevated_button.dart';
+import 'package:ods_flutter_demo/ui/components/buttons/component_button_customization.dart';
+import 'package:ods_flutter_demo/ui/main_app_bar.dart';
+
+import '../utilities/customization_bottom_sheet.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_elevated_button.dart:9]: prefer full path "import package:..."

File: app/lib/ui/components/buttons/component_elevated_button.dart [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    final ButtonCustomizationState? customizationState =
+        ButtonCustomization.of(context);
+
+    return Semantics(
+      header: true,
+      child: Padding(
+        padding:
+            const EdgeInsets.only(left: 15, top: 25, right: 15, bottom: 15.0),
+        child: OdsElevatedButton(
+          title: 'BUTTON',
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_elevated_button.dart:60]: should localize this text ?

File: app/lib/ui/components/buttons/component_filled_button.dart [hunk 14]
```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_filled_button.dart:1]: rename the file removing component to_ have filled_button.dart

File: app/lib/ui/components/buttons/component_filled_tonal_button.dart [hunk 15]
```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_filled_tonal_button.dart:1]: rename the file removing component to_ have filled_tonal_button.dart

File: app/lib/ui/components/buttons/component_filled_tonal_button.dart [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_gen/gen_l10n/ods_flutter_app_localizations.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:ods_flutter/components/button/ods_filled_tonal_button.dart';
+import 'package:ods_flutter_demo/ui/components/buttons/component_button_customization.dart';
+import 'package:ods_flutter_demo/ui/components/utilities/customization_bottom_sheet.dart';
+import 'package:ods_flutter_demo/ui/main_app_bar.dart';
+
+class ComponentFilledTonalButtons extends StatefulWidget {
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_filled_tonal_button.dart:10]: Always the same code for all buttons, it is possible to mutualize ?

File: app/lib/ui/components/buttons/component_outlined_button.dart [hunk 17]
```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_outlined_button.dart:1]: rename the file removing component to_ have outlined_button.dart

File: app/lib/ui/components/buttons/component_text_button.dart [hunk 18]
```diff
@@ -0,0 +1,106 @@
+import 'package:flutter/material.dart';
```
Comment (ludovic35) [inline app/lib/ui/components/buttons/component_text_button.dart:1]: rename the file removing component to_ have text_button.dart

File: library/lib/components/button/ods_elevated_button.dart [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+  final String title;
+
+  /// Specifies whether the button should expand to full screen width.
+  final bool fullScreenWidth;
+
+  /// The optional button's icon.
+  final Widget? icon;
+
+  /// The action to be executed when the button is pressed.
+  final Function()? onPressed;
```
Comment (ludovic35) [inline library/lib/components/button/ods_elevated_button.dart:32]: prefer VoidFunction

File: library/lib/components/button/ods_text_button.dart [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+  final Widget? icon;
+
+  /// The action to be executed when the button is pressed.
+  final Function()? onPressed;
+
+  @override
+  State<OdsTextButton> createState() => _OdsTextButtonState();
+}
+
+class _OdsTextButtonState extends State<OdsTextButton> {
```
Comment (ludovic35) [inline library/lib/components/button/ods_text_button.dart:38]: is ti possible to mutualize the code for all buttons ?

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: NOTICE.txt [hunk 21]
```diff
@@ -12,6 +12,7 @@ ods-flutter/app/assets/ic_about_info.svg
 ods-flutter/app/assets/ic_components_atom.svg
 ods-flutter/app/assets/ic_guidelines_dna.svg
 ods-flutter/app/assets/ic_modules_molecule.svg
+ods-flutter/app/assets/il_buttons.svg
```
Comment (ludovic35) [inline NOTICE.txt:15]: Update Notice.txt (add this asset in list)

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: app/CHANGELOG.md (hunk 9 above)
Comment (Tayebsed93) [reply app/CHANGELOG.md]: done

File: app/lib/ui/components/buttons/component_button_customization.dart (hunk 10 above)
Comment (Tayebsed93) [reply app/lib/ui/components/buttons/component_button_customization.dart]: done

File: app/lib/ui/components/buttons/component_elevated_button.dart (hunk 11 above)
Comment (Tayebsed93) [reply app/lib/ui/components/buttons/component_elevated_button.dart]: done

File: NOTICE.txt (hunk 21 above)
Comment (Tayebsed93) [reply NOTICE.txt:15]: sorry can you specify what is expected?

File: library/lib/components/button/ods_text_button.dart (hunk 20 above)
Comment (Tayebsed93) [reply library/lib/components/button/ods_text_button.dart:38]: I have not found a solution to mutualize the code for all buttons

File: NOTICE.txt (hunk 21 above)
Comment (ludovic35) [reply NOTICE.txt:15]: No sorry, it's ok

Comment (Tayebsed93) [reply NOTICE.txt:15]: ok

File: app/CHANGELOG.md [hunk 22]
```diff
@@ -15,8 +15,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Review the string in the app to use localization keys ([#129](https://github.com/Orange-OpenSource/ods-flutter/issues/129))
 - Add `OdsVerticalImageFirstCard` widget demonstration ([#8](https://github.com/Orange-OpenSource/ods-flutter/issues/8))
 - Add recipes Json file and use it ([#123](https://github.com/Orange-OpenSource/ods-flutter/issues/123))
-- Add customization bottom sheet to configure components ([#148](https://github.com/Orange-OpenSource/ods-flutter/issues/148)))
+- Add customization bottom sheet to configure components ([#148](https://github.com/Orange-OpenSource/ods-flutter/issues/148))
 - Add component navigation bar ([#23](https://github.com/Orange-OpenSource/ods-flutter/issues/23))
+- Add customization bottom sheet to configure components ([#148](https://github.com/Orange-OpenSource/ods-flutter/issues/148))
```
Comment (ludovic35) [inline app/CHANGELOG.md:20]: remove this line

**Review by ludovic35 — CHANGES_REQUESTED** [review]

**Review by ludovic35 — APPROVED** [review]

**Pull request merged**